django/views/decorators/debug.py

```diff
@@ -33,13 +38,42 @@ def my_function()
         )
 
     def decorator(func):
-        @wraps(func)
-        def sensitive_variables_wrapper(*func_args, **func_kwargs):
+        if iscoroutinefunction(func):
+
+            @wraps(func)
+            async def sensitive_variables_wrapper(*func_args, **func_kwargs):
+                return await func(*func_args, **func_kwargs)
```

Comment:
Oops, I missed this earlier:

These lines should be replaced with:
```
sensitive_variables_wrapper = func
```

I'll follow up on https://github.com/django/django/pull/16636

Reply:
I don't think it's a bug, but agree to change it as a cleanup.

Reply:
Ref https://github.com/django/django/pull/16636#issuecomment-1604755314, I would call this a bug or at least an edge case that needs to be handled